homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,335 @@
+        import github
+
+        self._github = github
+
+        self.setup_error = False
+
+        try:
+            if server_url is not None:
+                server_url += "/api/v3"
+                if access_token is not None:
+                    self._github_obj = github.Github(
+                        access_token, base_url=server_url)
+                elif username is not None and password is not None:
+                    self._github_obj = github.Github(
+                        username, password, base_url=server_url)
+            else:
+                if access_token is not None:
+                    self._github_obj = github.Github(access_token)
+                elif username is not None and password is not None:
+                    self._github_obj = github.Github(username, password)
+
+            self.repository_path = repository[CONF_PATH]
+
+            repo = self._github_obj.get_repo(self.repository_path)
+        except self._github.BadCredentialsException as err:
+            _LOGGER.error("Bad Credentials for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.UnknownObjectException as err:
+            _LOGGER.error("UnknownObjectException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.RateLimitExceededException as err:
+            _LOGGER.error("RateLimitExceededException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.BadAttributeException as err:
+            _LOGGER.error("BadAttributeException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.TwoFactorException as err:
+            _LOGGER.error("TwoFactorException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+        except self._github.GithubException as err:
+            _LOGGER.error("GithubException for %s: %s",
+                          self.repository_path, err)
+            self.setup_error = True
+
+        if self.setup_error is True:
+            return
+
+        if CONF_NAME in repository:
+            self.name = repository[CONF_NAME]
+        else:
+            self.name = repo.name
+
+        self.update = Throttle(interval)(self._update)
+
+        self.stargazers = None
+        self.topics = None
+        self.branches = None
+        self.open_issues = None
+        self.open_pull_requests = None
+        self.last_commit = None
+        self.latest_release = None
+        self.available = False
+
+    def _update(self):
+        """Update GitHub Sensor."""
+        try:
+            repo = self._github_obj.get_repo(self.repository_path)
+
+            self.stargazers = repo.stargazers_count
+
+            self.topics = json.dumps(repo.get_topics())
```

Comment:
This is a lot of information. What is the use case for this?

Reply:
None that I can think of off the top of my head. I suppose I can remove this to reduce the calls to the API.